internal/status/gql_reporter.go

```diff
@@ -38,12 +46,16 @@ func (r *GraphQLStatusReporter) ReportDeploymentStartup(ctx context.Context) (bo
 }
 
 func (r *GraphQLStatusReporter) ReportDeploymentShutdown(ctx context.Context) (bool, error) {
-	r.log.Debugf("Reporting deployment shutdown for ID: %s", r.gql.DeploymentID)
+	r.log.WithFields(logrus.Fields{
+		"deploymentID":    r.gql.DeploymentID,
+		"resourceVersion": r.getResourceVersion(),
+	}).Debugf("Reporting deployment shutdown...")
 	var mutation struct {
-		Success bool `graphql:"reportDeploymentShutdown(id: $id)"`
+		Success bool `graphql:"reportDeploymentShutdown(id: $id, resourceVersion: $resourceVersion)"`
 	}
 	variables := map[string]interface{}{
-		"id": graphql.ID(r.gql.DeploymentID),
+		"id":              graphql.ID(r.gql.DeploymentID),
+		"resourceVersion": r.getResourceVersion(),
 	}
 	if err := r.gql.Cli.Mutate(ctx, &mutation, variables); err != nil {
```

Comment:
What will happen when we fail with wrong resource version? Shouldn't we try to fetch Deployment (outside of `interval`) and try again?

Reply:
Yes, I agree, I'll add retry for that 👍